### PR TITLE
start new docker build

### PR DESCRIPTION
**Pull request by robkooper**
update build process
- fixes ed

## Review Time Estimate

- [x] Immediately
- [ ] Within one week
- [ ] When possible

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [x] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/docker.yml
Comment (infotroph) [inline .github/workflows/docker.yml]:
```suggestion
            R_VERSION=${{ github.event.inputs.r_version }}
```

File: docker/base/Dockerfile
Comment (infotroph) [inline docker/base/Dockerfile:30]: I'm glad to see this go, but I don't immediately see what replaces it -- are github tokens no longer needed inside this image, or is there now a cleaner way of passing them around?

Comment (infotroph) [reply docker/base/Dockerfile:30]: (Or am I asking too soon about a temporary change?)

Comment (robkooper) [reply docker/base/Dockerfile:30]: I have not needed the GITHUB_PAT, lets see what happens. I'll create an issue so we can keep track in case we need it.

File: docker/web/Dockerfile
Comment (infotroph) [inline docker/web/Dockerfile:2]: I bet you considered this already, but to confirm: Is deleting better than replacing with `LABEL org.opencontainers.image.authors="kooper@illinois.edu"`?

File: .github/workflows/docker.yml
Comment (infotroph) [inline .github/workflows/docker.yml]: Remove or uncomment?

File: .github/workflows/docker.yml
Comment (infotroph) [inline .github/workflows/docker.yml:117]: Each image's job has steps that are identical or close to it, correct? If so, I think we could likely save a fair amount of duplication by encapsulating these steps in a called workflow (see e.g. how check.yml/test.yml/sipnet.yml are used by ci.yml and ci-weekly.yml). 

(Probably just as easily done in a separate PR, so no need to hold this one up)

File: docker/base/Dockerfile
Comment (infotroph) [reply docker/base/Dockerfile:30]: And just to check: docker.depends/Dockerfile still contains the same `mount=type=secret` dance; is that intentional or should it go too?

File: models/ed/Dockerfile
Comment (infotroph) [inline models/ed/Dockerfile:13]: s/as/AS/ on line 7 above

File: docker-compose.yml
Comment (infotroph) [inline docker-compose.yml:74]: 👍

File: .github/workflows/docker.yml
Comment (infotroph) [inline .github/workflows/docker.yml:22]: Should devel (currently built weekly) and 4.4 (currently built daily) be added here? 

Is this approach enough faster that we no longer need to limit some builds to weekly?

File: .github/workflows/docker.yml
Comment (infotroph) [inline .github/workflows/docker.yml:35]: Am I right this step will completely replace depends.yml? How promptly? (I'm guessing we should delete it as soon as this is in production and confirmed to be behaving right?)

File: .github/workflows/docker.yml
Comment (infotroph) [inline .github/workflows/docker.yml:28]: Is the `|| '4.1'` needed here? I was assuming the `default: "4.1"` on line 21 would ensure a value, but haven't tested my assumption.

**Review by infotroph — APPROVED** [review]

File: docker/base/Dockerfile
Comment (robkooper) [reply docker/base/Dockerfile:30]: that should go as well 🤞

File: docker/web/Dockerfile
Comment (infotroph) [reply docker/web/Dockerfile:2]: Conclusion from further discussion in Slack: Yes, the existing MAINTAINER tags don't really line up well with who is actually maintaining each package, so better to delete them now and add more informative ones as time allows.

File: .github/workflows/docker.yml
Comment (infotroph) [reply .github/workflows/docker.yml:117]: Saving this for later as #3374

File: .github/workflows/docker.yml
Comment (infotroph) [reply .github/workflows/docker.yml:22]:
```suggestion
          - 4.3
          - 4.4
          - devel
```

Comment (mdietze) [reply .github/workflows/docker.yml:22]: Is this extending the set of R versions for just the Docker build, or for all the GH Actions? The other actions are currently running 4.1 and 4.2 with the former being the required version. If this is an overall change, it's outside the scope of the PR. If this is a Docker specific change, then I don't think we should be pushing a bunch of untested versions.

File: .github/workflows/docker.yml
Comment (robkooper) [reply .github/workflows/docker.yml:35]: correct, this will automatically build a new depends image for all builds. Once this is merged in, and we have a few builds going it would be good to see how well the caching works as well.

File: .github/workflows/docker.yml
Comment (robkooper) [reply .github/workflows/docker.yml:28]: the 4.1 is needed because the inputs are ONLY set when using a dispatch. :(

File: .github/workflows/docker.yml
Comment (robkooper) [reply .github/workflows/docker.yml:22]: this is just for the docker build. The nice thing is that the default is 4.1 but you can either use a curl or from a pull down test with others as well. Once we decide to change the default we can do that here as well, and when we want test with all other versions.

File: .github/workflows/docker.yml
Comment (robkooper) [reply .github/workflows/docker.yml:35]: Also this will remove the need to do a /build you can now trigger it manually

**Pull request merged**
